src/fcc_bridge_node_ros.cpp

```diff
@@ -412,19 +385,74 @@ void FCCBridgeNode::safety_limits_cb(const interfaces::msg::SafetyLimits &msg) {
 
     RCLCPP_INFO(this->get_ros_interface_logger(), "Set safety limits");
 
+    // Disable safety limits subscriber
+    this->safety_limits_subscriber.reset();
+
     // Go into WAITING_FOR_ARM state
     this->set_internal_state(INTERNAL_STATE::WAITING_FOR_ARM);
 }
 
+void FCCBridgeNode::control_cb(const interfaces::msg::Control &msg) {
+    RCLCPP_DEBUG(this->get_ros_interface_logger(),
+                 "Got new control message with target: %s and active: %s",
+                 msg.target_id.c_str(), msg.active ? "true" : "false");
+
+    if (common_lib::node_names::VALID_CONTROL_NODE_NAMES.find(msg.target_id) ==
+        common_lib::node_names::VALID_CONTROL_NODE_NAMES.end()) {
+        if (this->is_airborne()) {
+            // UAV is airborne. RTH is triggered
+            RCLCPP_ERROR(this->get_ros_interface_logger(),
+                         "Received a control message with an invalid "
+                         "target id %s! UAV is airborne. Triggering RTH...",
+                         msg.target_id.c_str());
+            this->trigger_rth();
+            return;
+        } else {
+            // UAV is on ground exiting
+            RCLCPP_FATAL(this->get_ros_interface_logger(),
+                         "Received a control message with an invalid "
+                         "target id %s! UAV is on ground. Exiting...",
+                         msg.target_id.c_str());
+            this->exit_process_on_error();
+        }
+    }
+
+    if (msg.active) {
+        if (this->active_node.has_value()) {

```

Comment:
Are you doing this as the fcc_bridge is more important in a way than the other nodes and we are therefore expecting an activation of the fcc_bridge should also work if another node is also (maybe incorrectly) active?

Reply:
The fcc_bridge is special in a way, that it does not receive control from mission control. Once it is active MC knows that it can send the safety limits and from there on the fcc_bridge is always active until an rth is triggered.
What it is doing here is recoding which node is currently active to then check that only the active node sends commands.
The fcc_bridge does not participate in the regular control pass around.

Reply:
Allright, got it thanks.